### DISCUSSION TITLE
Check if molecule has a query (without iteration)?

### DISCUSSION THREAD

**Discussion by kienerj** [rdkit/rdkit#6408] [Q&A] [answered]
Is there the possibility to check if a Mol instance in python contains an type of query with out iterating all atoms and bonds and call `HasQuery()`?

I don't see that method on Mol class level and the type is always `rdkit.Chem.rdchem.Mol` also when creating from SMARTS.

**Comment by ptosco** [comment, marked as answer]
@kienerj It looks like there isn't - adding a query on an atom or bond does not set any special flag on the parent molecule.
You either need to set a property yourself on the molecule if you know it has query features (e.g., because you generated it with `Chem.MolFromSmarts()` or manually added a query) or use a function such as
```
def has_query(mol):
    return (any(mol.GetAtomWithIdx(i).HasQuery() for i in range(mol.GetNumAtoms())) or
            any(mol.GetBondWithIdx(i).HasQuery() for i in range(mol.GetNumBonds())))
```
which will do the least amount of work needed.
